src/LocaleHelper.cpp

```diff
@@ -64,6 +64,12 @@ std::locale createUtf8Locale()
     // Don't change the locale on OSX. Using "" fails with 'locale::facet::_S_create_c_locale name not valid'
     return LocaleHelper::getBfsDefaultLocale();
 }
+#elif BOOST_OS_HAIKU
+std::locale createUtf8Locale()
+{
+    // Don't change the locale on Haiku. Using "" fails with 'locale::facet::_S_create_c_locale name not valid'
```

Comment:
Did you test this? Using `""` should work to get the system locale. Only apple is known to not give a sh*t

Reply:
Yeah, this is required, same error on Haiku.